libs/langchain/langchain/chat_models/bedrock.py

```diff
@@ -70,18 +77,24 @@ def _generate(
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
     ) -> ChatResult:
-        provider = self._get_provider()
-        prompt = ChatPromptAdapter.convert_messages_to_prompt(
-            provider=provider, messages=messages
-        )
+        completion = ""
 
-        params: Dict[str, Any] = {**kwargs}
-        if stop:
-            params["stop_sequences"] = stop
+        if self.streaming:
+            for chunk in self._stream(messages, stop, run_manager, **kwargs):
+                completion += chunk.text
+        else:
+            provider = self._get_provider()
+            prompt = ChatPromptAdapter.convert_messages_to_prompt(
+                provider=provider, messages=messages
+            )
 
-        completion = self._prepare_input_and_invoke(
-            prompt=prompt, stop=stop, run_manager=run_manager, **params
-        )
+            params: Dict[str, Any] = {**kwargs}
+            if stop:
+                params["stop_sequences"] = stop
```

Comment:
as per existing implementation the `BedrockChat` model supports only anthropic. When support for other providers are added to the `BedrockChat` model class it can be brought it.